src/main/scala/xiangshan/mem/vector/VSplit.scala

```diff
@@ -430,53 +384,26 @@ abstract class VSplitBuffer(isVStore: Boolean = false)(implicit p: Parameters) e
     splitIdx := 0.U(flowIdxBits.W) // initialize flowIdx
     strideOffsetReg := 0.U
   }
-  // allocatedWire, only for freelist select next uop back-to-back
-  for (i <- 0 until bufferSize){
-    when(needCancel(i)){ // redirect
-      allocatedWire(i) := false.B
-    }.elsewhen(splitFinish && (activeIssue || inActiveIssue) && (i.U === selIdxReg)){ // finish
-      allocatedWire(i) := false.B
-    }.otherwise{
-      allocatedWire(i) := allocated(i)
-    }
-  }
   // allocated
-  for (i <- 0 until bufferSize){
-    when(needCancel(i)) { // redirect
-      allocated(i) := false.B
-    }.elsewhen(splitFinish && (activeIssue || inActiveIssue) && (i.U === selIdxReg)){ //dequeue
-      allocated(i) := false.B
-    }.elsewhen(doEnqueue && (i.U === enqIndex)){
-      allocated(i) := true.B
-    }
-  }
-  // freeMask
-  for (i <- 0 until bufferSize){
-    when(needCancel(i)) { // redirect
-      freeMask(i) := true.B
-    }.elsewhen(splitFinish && (activeIssue || inActiveIssue) && (i.U === selIdxReg)) { //dequeue
-      freeMask(i) := true.B
-    }.otherwise{
-      freeMask(i) := false.B
-    }
+  when(doEnqueue){ // if enqueue need to been cancelled, it will be false, so this have high priority
+    allocated := true.B
+  }.elsewhen(needCancel) { // redirect
+    allocated := false.B
+  }.elsewhen(splitFinish && (activeIssue || inActiveIssue)){ //dequeue
+    allocated := false.B
```

Comment:
If `needCancel` has higher priority, shouldn't it be like:

> when (needCancel) {
  ...
}.elsewhen (doEnqueue) {
  ...
}.elsewhen (splitFinish && (activeIssue || inActiveIssue)) {
  ...
}

Reply:
`needCancel` means entry need to be flush that have enter.  And `doEnqueue = true` means enter request will be not flush and will be latch next cycle, which indicate `doEnqueue` alrady do Cancel checker. So, doEnqueue have high priority.

ps: The logic of `doEnqueue`  is `doEnqueue := io.in.valid && !io.in.bits.uop.robIdx.needFlush(io.redirect)`, So if `doEnqueue` is true, entry need to be valid.